### ISSUE TITLE
object-shorthand: Allow different rules for methods and properties

### ISSUE THREAD

**Issue by Ariane-B** [eslint/eslint#14245]
**The version of ESLint you are using.**
7.21.0

**The problem you want to solve.**
We'd like to solve inconsistencies to objects where some properties are shorthand, while others are not. My first idea was to use the `consistent` option of the `object-shorthand` rule. The problem with that is that it gives off an error on our shorthand methods, which we consider always okay no matter the circumstances.

✔ For instance, under the ideal rule, these would be correct:

```javascript
let myObject = {
  firstProperty: firstValue,
  secondProperty: secondValue,
  firstMethod(arg) {},
}

let myObject = {
  firstProperty,
  secondProperty,
}
```

❌ But these would not be correct:

```javascript
let myObject = {
  firstProperty,
  secondProperty: secondValue,
}

let myObject = {
  firstProperty: firstValue,
  secondProperty: secondValue,
  firstMethod(arg) {},
  secondMethod: function(arg) {},
}
```

**Your take on the correct solution to problem.**

It would be nice if it were possible to say that properties should be consistent between them and that methods should be consistent between them. For instance, instead of having a global string value, there would only be an options object, that would look like this:

```javascript
{
  // Contains rules that affect methods between themselves
  methods: {
    base: 'consistent',
    allowArrows: 'shorthand'
  },
  // Contains rules that affect properties between themselves
  properties: {
    base: 'consistent',
    allowQuotes: 'long-form',
  },
  // Contains rules that affect both methods and properties in a mixed manner
  all: {
    // No rules at all binding properties and methods together
    base: 'ignore'
  },
}
```

**Are you willing to submit a pull request to implement this change?**
Sorry, no. I don't think I'd be able to.

**Comment by mdjermanovic** [comment]
Hi @Ariane-B, thanks for the issue!

The team has made the difficult decision to freeze stylistic rules (please find more details in [this blog post](https://eslint.org/blog/2020/05/changes-to-rules-policies)), so we won't be adding any new options to the `object-shorthand` rule. You can make a custom rule or publish a [plugin](https://eslint.org/docs/developer-guide/working-with-plugins).

I'm closing this issue per our [rules policies](https://github.com/eslint/eslint#stylistic-rule-updates).

**Comment by Ariane-B** [comment]
Hi! Thanks for the info. I'm disappointed but I understand. Have a nice day!
